### PR TITLE
main: Update ginkgo version to v2

### PR DESCRIPTION
**Pull request by gururajsh**
Thank you for contributing to the CF CLI! Please read the following:

* Please make sure you have implemented changes in line with the [contributing guidelines](https://github.com/cloudfoundry/cli/blob/main/.github/CONTRIBUTING.md)
* We're not allowed to accept any PRs without a signed CLA, no matter how small.
If your contribution falls under a company CLA but your membership is not public, expect delays while we confirm.
* All new code requires tests to protect against regressions.
* Contributions must be made against the appropriate branch. See the [contributing guidelines](https://github.com/cloudfoundry/cli/blob/main/.github/CONTRIBUTING.md)
* Contributions must conform to our [style guide](https://github.com/cloudfoundry/cli/wiki/CLI-Product-Specific-Style-Guide). Please reach out to us if you have questions.

## Where this PR should be backported?

- [x] [main](https://github.com/cloudfoundry/cli/tree/main) - all changes should by default start here
- [x] [v8](https://github.com/cloudfoundry/cli/tree/v8)
- [x] [v7](https://github.com/cloudfoundry/cli/tree/v7)

## Description of the Change

CLI was using ginkgo version v1 which is outdated for a while now. This change updates the ginkgo version to v2.

## Why Is This PR Valuable?

Updates the ginkgo version so that there is on going support for this dependency.

## Applicable Issues

None

## How Urgent Is The Change?

Not urgent

## Other Relevant Parties

None

### CODE REVIEW

File: .github/workflows/tests-integration-reusable.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
     runs-on: ${{ inputs.os }} 
 
     steps:
-    - uses: LouisBrunner/checks-action@v2.0.0
```
Comment (moleske) [inline .github/workflows/tests-integration-reusable.yml:35]: just checking it is intentional that is being downgraded

**Review by moleske — COMMENTED** [review]
Very excited to see this!

File: .github/workflows/tests-integration-reusable.yml (hunk 1 above)
Comment (gururajsh) [reply .github/workflows/tests-integration-reusable.yml]: Good catch!! Fixed it, it wasn't intentional.

File: .github/workflows/test-shepherd.yml
Comment (a-b) [inline .github/workflows/test-shepherd.yml:1]: This looks unrelated to the Ginkgo v2

Comment (gururajsh) [reply .github/workflows/test-shepherd.yml:1]: Thanks!! removed it.

**Review by a-b — APPROVED** [review]
Reviewed with @dalvarado
